### PR TITLE
Refactor Tx body creation and signing

### PR DESCRIPTION
**Pull request by nazrhom**

### CODE REVIEW

**Comment by nazrhom** [comment]
Separating this will be useful for the second part of #203

File: src/HydraAuction/Tx/Escrow.hs [hunk 1]
```diff
@@ -136,6 +136,7 @@ announceAuction terms = do
     autoSubmitAndAwaitTx $
       AutoCreateParams
         { authoredUtxos = [(sellerSk, utxoWithLotNFT <> sellerMoneyUtxo)]
+        , signers = [sellerSk]
```
Comment (uhbif19) [inline src/HydraAuction/Tx/Escrow.hs]: Why add extra signer here, if it is already in `authoredUtxos`?

File: src/HydraAuction/Tx/Common.hs [hunk 2]
```diff
@@ -318,7 +320,18 @@ toSlotNo ptime = do
   either (error . show) return $ slotFromUTCTime timeHandle utcTime
 
 autoCreateTx :: AutoCreateParams -> Runner Tx
-autoCreateTx (AutoCreateParams {..}) = do
+autoCreateTx params = do
+  body <- autoCreateTxBody params
+  pure $ signTx (signers params) body
```
Comment (uhbif19) [inline src/HydraAuction/Tx/Common.hs]: It should use `authoredUtxos` signers as well.

**Review by uhbif19 — CHANGES_REQUESTED** [review]

File: src/HydraAuction/Tx/Common.hs (hunk 2 above)
Comment (nazrhom) [reply src/HydraAuction/Tx/Common.hs]: Can you explain how we are using `authoredUtxos`? We use it to fill `TxExtraKeyWitnesses`, but that requires a VKey, we are passing in an SKey and every user will only have access to their own SKey.

I can not find the official docs on `TxExtraKeyWitnesses`, but my understanding is that you specify extra VKeys that will need to sign the tx. For example, the bidder could construct a Tx which spends the collateral of the delegate, and would need to add the delegate VKey as extra key witness? Maybe I am getting it wrong though, could you also clarify what that field means in the tx then?

Comment (uhbif19) [reply src/HydraAuction/Tx/Common.hs]: > Can you explain how we are using authoredUtxos?

It is bunch of UTxOs paired with keys to sign them. That API makes clear which UTxO uses which keys and prevents error on missing signing witness or signing.

Not sure that I fully understand your question thought.

File: src/HydraAuction/Tx/Common.hs [hunk 3]
```diff
@@ -294,6 +294,8 @@ scriptUtxos script terms = do
 
 data AutoCreateParams = AutoCreateParams
   { authoredUtxos :: [(SigningKey PaymentKey, UTxO)]
+  , signers :: [SigningKey PaymentKey]
```
Comment (uhbif19) [inline src/HydraAuction/Tx/Common.hs]: Can you name it `additionalSigners` or something like that? Cuz currently is not clear that there is an other thing that signs.

Also we can rename `authoredUtxos` into `signedUtxos`.

**Review by uhbif19 — APPROVED** [review]

**Pull request merged**